storage/pebble/blocks_provider_test.go

```diff
@@ -0,0 +1,353 @@
+package pebble
+
+import (
+	"testing"
+
+	"github.com/goccy/go-json"
+	"github.com/onflow/flow-evm-gateway/config"
+	"github.com/onflow/flow-evm-gateway/models"
+	"github.com/onflow/flow-evm-gateway/storage"
+	"github.com/onflow/flow-evm-gateway/storage/mocks"
+	"github.com/onflow/flow-go-sdk"
+	evmTypes "github.com/onflow/flow-go/fvm/evm/types"
+	flowGo "github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/go-ethereum/common"
+	"github.com/onflow/go-ethereum/eth/tracers"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	// this import is needed for side-effects, because the
+	// tracers.DefaultDirectory is relying on the init function
+	_ "github.com/onflow/go-ethereum/eth/tracers/native"
+)
+
+func TestOnBlockReceived(t *testing.T) {
+
+	t.Run("without latest block", func(t *testing.T) {
+		blocks := setupBlocksDB(t)
+		blocksProvider := NewBlocksProvider(blocks, flowGo.Emulator, nil)
+
+		block := mocks.NewBlock(1)
+		err := blocksProvider.OnBlockReceived(block)
+		require.NoError(t, err)
+	})
+
+	t.Run("with nil block", func(t *testing.T) {
+		blocks := setupBlocksDB(t)
+		blocksProvider := NewBlocksProvider(blocks, flowGo.Emulator, nil)
+
+		err := blocksProvider.OnBlockReceived(nil)
```

Comment:
Is this a valid case? I think it's the caller's responsibility to ensure the block is not nil.

Reply:
Good point. I've removed this in https://github.com/onflow/flow-evm-gateway/pull/632/commits/3cbf6ae8024c4d399f7d86d5a0f2840108fba573.